src/pastefromword.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module pastefromword/pastefromword
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import DocumentFragment from '@ckeditor/ckeditor5-engine/src/view/documentfragment';
+
+import { extractBody, bodyToView, extractStyles } from './filters/common';
+import { paragraphsToLists } from './filters/list';
+
+/**
+ * This plugin handles content pasted from Word and transforms it (if necessary)
+ * to format suitable for editor {@link module:engine/model/model~Model}.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class PasteFromWord extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'PasteFromWord';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+		const document = editor.editing.view.document;
+
+		this.listenTo( document, 'clipboardInput', ( evt, data ) => {
+			const html = data.dataTransfer.getData( 'text/html' );
+
+			if ( isWordInput( html ) ) {
+				evt.stop();
+
+				const normalizedInput = this._normalizeWordInput( html, editor );
+
+				editor.plugins.get( 'Clipboard' ).fire( 'inputTransformation', { content: normalizedInput } );
+			}
+		} );
+	}
+
+	/**
+	 * Normalizes input pasted from Word to format suitable for editor {@link module:engine/model/model~Model}.
+	 *
+	 * @private
+	 * @param {module:core/editor/editor~Editor} editor Editor instance.
+	 * @param {String} input Word input.
+	 * @returns {module:engine/view/node~Node|module:engine/view/documentfragment~DocumentFragment} view Normalized input.
+	 */
+	_normalizeWordInput( input, editor ) {
```

Comment:
Is this function necessary? The listener, in which it's used is fairly short, so it doesn't seem so.

Reply:
TBH, I have extracted it so it can be easily unit tested, that was the main reason. We may inline it but then tests will have to listen to `inputTransformation` event to get the normalized data. Since those unit tests use only `Paste from Word` and `Clipboard` plugin there is rather a small chance that something in `Clipboard` plugin will change that can interfere with the normalized input before it is validated in a test 🤔